### PR TITLE
hydrogen-web: init at 0.4.0

### LINKED ISSUE #141586: hydrogen-web
**Project description**
A minimal Matrix chat client, focused on performance, offline functionality, and broad browser support.

**Metadata**

* homepage URL: https://hydrogen.element.io/
* source URL: https://github.com/vector-im/hydrogen-web/
* license: Apache
* platforms: Browser

Deer matrix-team maintainers:
@ma27
@fadenb
@mguentner
@ekleog
@ralith
@mjlbach
@dandellion
@sumnerevans

### PR DESCRIPTION
**Pull request by linsui**
###### Description of changes
Closes https://github.com/NixOS/nixpkgs/issues/141586

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
How much is this just copied from element? Can we make this a variant of element without copying all the instructions? Maybe overriding source is already enough?

**Comment by linsui** [comment]
This is mostly copied from element removing some unused part. But they are two totally different projects so I thought it makes no sense to merge them. They just both use yarn.

File: pkgs/applications/networking/instant-messengers/hydrogen-web/unwrapped.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    cp -R target $out
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Lightweight matrix client with legacy and mobile browser support";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/hydrogen-web/unwrapped.nix:59]: homepage is missing

Comment (linsui) [reply pkgs/applications/networking/instant-messengers/hydrogen-web/unwrapped.nix:59]: Done.

**Review by drupol — APPROVED** [review]

**Pull request merged**